packages/amplify-cli-utils/src/telemetry.js

```diff
@@ -0,0 +1,107 @@
+import fs from 'fs-extra';
+import loadConfig from '@axway/amplify-config';
+import path from 'path';
+import snooplogg from 'snooplogg';
+import { createRequestOptions } from './request';
+import { Telemetry } from '@axway/amplify-sdk';
+import * as environments from './environments';
+import * as locations from './locations';
+
+const { warn } = snooplogg('amplify-cli-utils:telemetry');
+const telemetryCacheDir = path.join(locations.axwayHome, 'axway-cli', 'telemetry');
+let telemetryInst = null;
+
+/**
+ * If telemetry is enabled, writes the anonymous event data to disk where it will eventually be
+ * sent to the Axway platform.
+ *
+ * @param {Object} payload - the telemetry event payload.
+ * @param {Object} [opts] - Various options to pass into the `Telemetry` instance.
+ */
+export function addEvent(payload, opts) {
+	init(opts);
+	// eslint-disable-next-line no-unused-expressions
+	telemetryInst?.addEvent(payload);
+}
+
+/**
+ * If telemetry is enabled, writes the anonymous crash event to disk where it will eventually be
+ * sent to the Axway platform.
+ *
+ * @param {Object} payload - the telemetry event payload.
+ * @param {Object} [opts] - Various options to pass into the `Telemetry` instance.
+ */
+export function addCrash(payload, opts) {
+	init(opts);
+	// eslint-disable-next-line no-unused-expressions
+	telemetryInst?.addCrash(payload);
+}
+
+/**
+ * Checks if telemetry is enabled, then if it is, creates the telemetry instance and registers the
+ * send handler.
+ *
+ * @param {Object} [opts] - Various options.
+ * @param {String} [opts.appGuid] - The platform registered app guid.
+ * @param {String} [opts.appVersion] - The app version.
+ * @param {Config} [opts.config] - The Axway CLI config object.
+ * @param {String} [opts.env] - The environment name.
+ * @param {String} [opts.url] - The platform analytics endpoint URL.
+ * @returns {Telemetry}
+ */
+export function init(opts = {}) {
+	try {
+		if (telemetryInst) {
+			return;
```

Comment:
maybe return found telemetryInst here too so you will be able to chain it ( init(opts)?.addEvent(payload) ) in "addEvent" and "addCrash" methods?